compiler/codegen/CodeGenPrep.cpp

```diff
@@ -103,15 +103,11 @@ OMR::CodeGenerator::lowerTreesPreChildrenVisit(TR::Node * parent, TR::TreeTop *
 
       if (secondChild->getOpCode().isLoadConst())
          {
-         bool isInteger = (parent->getOpCodeValue() == TR::imul) ? true : false;
-         int64_t value;
-         if (isInteger)
-            value = secondChild->getInt();
-         else
-            value = secondChild->getLongInt();
+         bool isInteger = (parent->getOpCodeValue() == TR::imul);
```

Comment:
This variable seems poorly named. Both `imul` and `lmul` are integer operations. This seems to be checking whether the operation is 32-bit or 64-bit. We should also use the IL property to check the width rather than explicit opcode check. Do you agree?

Reply:
Yeah, that's good idea. I'll push a fix.

Reply:
Fixed in https://github.com/eclipse/omr/pull/4607/commits/a01021769886bfe16f9d72241068e834ec53e705